### PR TITLE
Merge elements and objects as components

### PR DESCRIPTION
**Pull request by philschanely**
## Description

This PR reorganizes the docs site and `sage-assets` style directories to combine Elements and Objects into a single list of Components. Supporting content and generators are updated as well (`sage_component` now generates all aspects of a component and `sage_rails_component` is deleted).

## Test notes

- Navigate the site to ensure component structure is integrated.
- Test the new component generator:
```
cd docs
rails generate sage_component MyComponent name details is_active
```

### Estimated impact
(LOW) Documentation updates. No impact on existing components.

### CODE REVIEW

**Review by monkeypox8 — APPROVED** [review]
👍🏼  LGTM! In my eyes, this change should help reduce mental overhead quite a bit.

Side note, any thoughts on a future state where we can display a long list of component "snippets"/thumbnails as a component landing page?

**Comment by monkeypox8** [comment]
Just noticed one more thing that can be addressed in a follow-up. The [Panels and Cards page](http://localhost:4000/pages/layout/panels_cards) still refers to Card and Panel Objects:

panel and card description page

**Comment by philschanely** [comment]
> Just noticed one more thing that can be addressed in a follow-up. The [Panels and Cards page](http://localhost:4000/pages/layout/panels_cards) still refers to Card and Panel Objects:
> 
> panel and card description page

Ah thanks! I will run a quick audit on "object" and "Object" in the site to see if there are any others.

File: docs/lib/generators/sage_component/templates/markup.html.erb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<div
+  class="sage-<%= file_name %> <%%= component.generated_css_classes %>"
```
[the remark this answers is not in the document]
Comment (philschanely) [reply docs/lib/generators/sage_component/templates/markup.html.erb:2]: Good point; these appear to be needed due to the generator and template inserting values.

**Review by pixelflips — APPROVED** [review]
A non-blocking nitpick but this is great. I wasn't sure how I felt about the combining, but after seeing it I think it's definitely an improvement. Nice work! 👍🏼

File: docs/app/views/examples/components/panel/_preview.html.erb [hunk 2]
```diff
@@ -48,11 +48,11 @@ Panels are rich containers that allow for grouping and organizing content detail
 
 - **Body**
   While there is no specific container for the body itself,
-  there are a variety of panel elements that can be used to create most panel configurations.
+  there are a variety of panel components that can be used to create most panel configurations.
   The simplest of these **Panel Block** `.sage-panel__block` (shown above)
   allows for open text or elements with no preset spacing is desired.
```
[the remark this answers is not in the document]
Comment (philschanely) [reply docs/app/views/examples/components/panel/_preview.html.erb:53]: Thanks @monkeypox8 . I believe `components` is a good fit here in that I do mean to refer to the actual Sage PanelX Components. But if you feel `elements` makes more sense I'm not opposed to it.

I am not sure I follow your second point... do you mind providing the suggested revision?

[the remark this answers is not in the document]
Comment (monkeypox8) [reply docs/app/views/examples/components/panel/_preview.html.erb:53]: Ah, I realize now the way I'd worded it was confusing. So, two possible edits that I had in mind:
```The simplest of these **Panel Block** `.sage-panel__block` (shown above) allows for open text or elements where no preset spacing is desired.```

or

```The simplest of these **Panel Block** `.sage-panel__block` (shown above) allows for open text or elements with no preset spacing desired.```

[the remark this answers is not in the document]
Comment (philschanely) [reply docs/app/views/examples/components/panel/_preview.html.erb:53]: Got it @monkeypox8 thank you! Fixing now.

**Pull request merged**